### ISSUE TITLE
TouchBarServer.app V1.4 crash when plug in iPad

### ISSUE THREAD

**Issue by moriwang** [bikkelbroeders/TouchBarDemoApp#54]
I tested the previous version, the touchbarserver works fine with iPad.
But the latest version will crash once I plug in my iPad opening Touch Bar App.

First I open both Touch Bar App and Touchbarserver in my Mac.
The app shows "Please connect this device to your Mac and start TouchBarServer..."
The server works on the top of screen.
Then I plug in my iPad, the server icon disappear and the iPad remains the same.
I can't open the server again until l unplug my iPad.

**Comment by robbertkl** [comment]
Sorry to hear this @moriwang, let's get to the bottom of this. Can you fetch us the crash log for TouchBarServer? You can find instructions on where to find crashlogs [here](https://marinersoftware.deskpro.com/kb/articles/how-to-retrieve-crash-logs-in-os-x).

**Comment by smartjx** [comment]
I got the same issue here. Please find my crashlog

[TouchBarServer_2016-11-16-222601_xiangs-iMac.crash.zip](https://github.com/bikkelbroeders/TouchBarDemoApp/files/594877/TouchBarServer_2016-11-16-222601_xiangs-iMac.crash.zip)

**Comment by Fetiz** [comment]
I get this status too.
After I plug my iPad Air 2, Touchbarserver app disappear on the menu bar. 
But the process still working and I can't reopen the touchbarserver app.
waiting for you fix this crash. the log is in the 
[TouchBarServer_2016-11-16-232343_MacBook-Pro.crash.zip](https://github.com/bikkelbroeders/TouchBarDemoApp/files/595036/TouchBarServer_2016-11-16-232343_MacBook-Pro.crash.zip)

**Comment by smartjx** [comment]
1.Open System Preferences from the Apple menu ().
2.Click the Security & Privacy icon.
3.Select the Privacy tab.
4.Select Accessibility from the list on the left side of the Privacy pane.
5.Click the lock icon in the lower left corner of the window and enter an admin account and password when prompted.
6. Add TouchBarServer and select the checkbox next to the name allow requesting access. 

Now it works!

**Comment by robbertkl** [comment]
After inspecting the crashlogs (thanks for those!) we've found an inconsistency/bug with certain special keyboard layouts. Can you tell me if the [latest release](https://github.com/bikkelbroeders/TouchBarDemoApp/releases/latest) works again?

@smartjx Your suggestion does not make sense, as the accessibility setting is no longer needed and also has nothing to do with the remote iOS Touch Bar clients.

**Comment by moriwang** [comment]
@robbertkl  It works. Thanks for your work!
